usaspending_api/search/v2/views/search.py

```diff
@@ -233,35 +234,34 @@ def post(self, request):
             queryset, model = spending_by_award_count(filters)
 
         if subawards:
-            queryset = queryset.values("award_type").annotate(category_count=Count("subaward_id"))
+            queryset = queryset.values("prime_award_type").annotate(category_count=Count("subaward_id"))
         elif model == "SummaryAwardView":
-            queryset = queryset.values("category").annotate(category_count=Sum("counts"))
+            queryset = queryset.values("type").annotate(category_count=Sum("counts"))
         else:
-            queryset = queryset.values("category").annotate(category_count=Count("category"))
-
-        categories = (
-            {
-                "contract": "contracts",
-                "idv": "idvs",
-                "grant": "grants",
-                "direct payment": "direct_payments",
-                "loans": "loans",
-                "other": "other",
-            }
-            if not subawards
-            else {"procurement": "subcontracts", "grant": "subgrants"}
-        )
-
-        category_name = "category" if not subawards else "award_type"
+            queryset = queryset.values("type").annotate(category_count=Count("category"))
 
         # DB hit here
         for award in queryset:
-            if award[category_name] is None:
-                result_key = "other" if not subawards else "subcontracts"
-            elif award[category_name] not in categories.keys():
-                result_key = "other"
+            if subawards:
+                if award["prime_award_type"] in assistance_type_mapping.keys():
+                    result_key = "subgrants"
+                else:
+                    result_key = "subcontracts"
             else:
-                result_key = categories[award[category_name]]
+                if award["type"] in contract_type_mapping.keys():
```

Comment:
One code-smell in python is implementing a case-statement-like if-elif conditional

A common pattern in place of the if-elif-elif...  is to use a dictionary. A new dict can be created from `all_award_types_mappings` like this:
```all_awards_types_to_category = {type_code: category for category, type_codes in all_award_types_mappings.items() for type_code in type_codes}```

Then the results can be combined using the dict lookup.